src/python/WMCore/WMSpec/Steps/Executor.py

```diff
@@ -177,6 +177,8 @@ def setCondorChirpAttrDelayed(self, key, value, compress=False, maxLen=5120):
         Util to call condor_chirp and publish the key/value pair
 
         """
+        # bytes object is not JSON serializable in Python3
+        value = decodeBytesToUnicode(value)
         if compress:
             value = zipEncodeStr(value, maxLen=maxLen)
```

Comment:
Hello! Sorry for commenting without being asked to :grimacing: . I have a suggestion and a comment.

Since py2 is working, i wouldn't change it. I know that it is unlikely to cause any problem, but we never know (and i super conservative ahaha).

Moreover, `zipEncodeStr()` accepts unicode and bytes as input, converts the unicode to bytes internally, then return bytes (`str` in py2, `bytes` in py3). I would move `value = decodeBytesToUnicode(value)` down a couple of lines, after `if compress: value = zipEncodeStr(value, maxLen=maxLen)`, outside of the if, otherwise if we set `compress` to `True` we have to decode bytes to unicode again. Do you agree?

Reply:
Haa, your input is always very welcomed, Dario! I'm glad you caught the terrible proposal I had initially made here ;)
I'm pushing more changes soon, considering your suggestions.

Reply:
Thanks Alan! :D